src/equisolve/module.py

```diff
@@ -0,0 +1,62 @@
+from abc import abstractmethod
+
+from typing import TypeVar
+
+from equistore import TensorMap
+
+# Workaround for typing Self with inheritance for python <3.11
+# see https://peps.python.org/pep-0673/
+TModule = TypeVar("TModule", bound="Module")
+
+try:
+    import torch
+    Module = torch.nn.Module
+except:
+    class Module:
+        @abstractmethod
+        def forward(self, X: TensorMap):
```

Comment:
`forward` in Torch can take an arbitrary number of positional and keyword parameters, should we allow for the same here? (I'm not sure what `@abstractmethod` does anyway?)

Reply:
Yes. 

`@abstractmethod` ensures that the method is implemented on construction of the object. The raise NotImplemented errors are actually not needed. I removed them. I had this just in my mind, because I had a case where I could not inherit from `ABCMetaclass`, and in this case `@abstractmethod` is actually doing nothing.

Reply:
ok, thanks!